pkg/api/v1/routes/handlers.go

```diff
@@ -119,6 +119,34 @@ func (r *Routes) serverConditionCreate(c *gin.Context) (int, *v1types.ServerResp
 	return r.conditionCreate(otelCtx, newCondition, serverID, facilityCode)
 }
 
+func (r *Routes) serverDelete(c *gin.Context) (int, *v1types.ServerResponse) {
+	otelCtx, span := otel.Tracer(pkgName).Start(c.Request.Context(), "Routes.serverDelete")
+	span.SetAttributes(attribute.KeyValue{Key: "serverId", Value: attribute.StringValue(id)})
+	defer span.End()
+
+	serverID, _ := uuid.Parse(c.Param("uuid"))
+	if err := r.fleetDBClient.DeleteServer(c.Request.Context(), serverID); err != nil {
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
+	}
+
+	// Do we need to
+	// publish/update condtion letting subscriber knows that this condition is no longer valid?
+	// or is repository.Delete already good enough?
```

Comment:
I would not touch any conditions that are in flight, because it will cause more problems than it solves. If someone deletes a server while a condition is being worked on, the fleetDB updates will fail, and that will probably cause the executing condition to fail as well. That's acceptable, in my view.

You could use `r.repository.GetActiveCondition` to make sure that the server was actually not being worked on. I think that's probably a better test. If there is an active condition on the server, fail the delete.